hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -103,14 +105,19 @@ protected void init(HoodieTableMetaClient metaClient, HoodieTimeline visibleActi
    * @param visibleActiveTimeline Visible Active Timeline
    */
   protected void refreshTimeline(HoodieTimeline visibleActiveTimeline) {
-    this.visibleCommitsAndCompactionTimeline = visibleActiveTimeline.getCommitsAndCompactionTimeline();
+    this.visibleCommitsAndCompactionTimeline = visibleActiveTimeline.getCommitsReplaceAndCompactionTimeline();
   }
 
   /**
    * Adds the provided statuses into the file system view, and also caches it inside this object.
    */
   protected List<HoodieFileGroup> addFilesToView(FileStatus[] statuses) {
     HoodieTimer timer = new HoodieTimer().startTimer();
+    final Map<String, Set<String>> partitionFileIdsToExclude = getFileIdsToExclude(visibleCommitsAndCompactionTimeline);
```

Comment:
Does this mean that we can never go back to querying the older file groups once they have been replaced ? Can you still do time-travel for insert-overwrite use-cases ?

Reply:
Yes, for time travel, consider this scenario:
t0 -> insert
t1 -> insert overwrite1
t2 -> insert overwrite2

If we set high watermark to t1 for time travel, visibleCommitTimeline would not have t2.commit, t2.replace. So file groups in t1 would still show as active file groups.

When we move to t2, visibleCommitTimeline will have t2 commit/replace. So file groups in t1 will not show as active

Reply:
[Test here ](https://github.com/apache/hudi/pull/1859/files#diff-288e80cb950f55ef8aa60eb5abcfdb23R1227) simulates rollback of replace instant. 

I can add another one by filtering timeline to move high watermark.